### DISCUSSION TITLE
[V5] Decimal field is returning as string

### DISCUSSION THREAD

**Discussion by victor-sc-92** [adonisjs/core#2764] [Help] [answered]
I create a price field on migration and defined it as number on model, but when it's serialized, returns a string.
This happens in other fields that is decimal too, how can i fix it?

```ts
export default class Provides extends BaseSchema {
  protected tableName = 'provides'

  public async up(): Promise<void> {
    this.schema.createTable(this.tableName, (table) => {
      table.uuid('id').primary()
      ...
      table.decimal('price', 12, 2)

      table.timestamp('created_at', { useTz: true })
      table.timestamp('updated_at', { useTz: true })
    })
  }

  public async down(): Promise<void> {
    this.schema.dropTable(this.tableName)
  }
}
```

```ts
export default class Provide extends BaseModel {
  public static selfAssignPrimaryKey = true

  @column({ isPrimary: true })
  public id: string

  ...

  @column()
  public price: number

 ...

  @column.dateTime({ autoCreate: true })
  public createdAt: DateTime

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  public updatedAt: DateTime

  @beforeCreate()
  public static assignUuid(provide: Provide): void {
    provide.id = uuid()
  }
}

```

**Comment by skrenek** [comment, marked as answer]
Some database drivers (Postgres, for example) will return decimals as strings due to possible loss of precision with floating point numbers.  If you're using Postgres, we've worked around this by adding the following to the register function of the app's AppProvider.  It will convert the strings to numbers before they hit the Adonis framework.  If you're not using Postgres, your mileage may vary, but hopefully this sends you in the right direction.

    public register () {
      const types = require('pg').types
      types.setTypeParser(types.builtins.NUMERIC, function (val) {
        return Number(val)
      })
    }

**Comment by arthurcvm** [comment]
In `start/kernel.ts` you import this:

```ts
import pg from 'pg'
```
and call to pg types:
```ts
pg.types.setTypeParser(20, 'text', parseInt)
```
